rest/data/project_settings.go

```diff
@@ -317,3 +318,53 @@ func handleGithubConflicts(pRef *model.ProjectRef, reason string) error {
 	}
 	return nil
 }
+
+// DeleteContainerSecrets deletes existing container secrets in the project ref
+// from the secrets storage service. This returns the remaining secrets after
+// deletion.
+func DeleteContainerSecrets(ctx context.Context, v cocoa.Vault, pRef *model.ProjectRef, namesToDelete []string) ([]model.ContainerSecret, error) {
+	catcher := grip.NewBasicCatcher()
+	var remaining []model.ContainerSecret
+	for _, secret := range pRef.ContainerSecrets {
+		if !utility.StringSliceContains(namesToDelete, secret.Name) {
+			remaining = append(remaining, secret)
+			continue
+		}
+
+		if secret.ExternalID != "" {
+			catcher.Wrapf(v.DeleteSecret(ctx, secret.ExternalID), "deleting container secret '%s' with external ID '%s'", secret.Name, secret.ExternalID)
```

Comment:
`remaining` isn't really used if there's an error, because it's an intermediate state. I changed it to check if the catcher has errors before returning the remaining secrets.